src/test/java/org/jenkinsci/plugins/envinject/EnvInjectBuildWrapperTest.java

```diff
@@ -61,47 +65,35 @@ public void injectFromScript() throws Exception {
     }
 
     @Test
-    public void testPropertiesContentCustomWorkspace() throws Exception {
-
-        String customWorkspaceValue = Hudson.getInstance().getRootPath().getRemote() + "/customWorkspace";
+    public void shouldPopulatePropertiesContentWithCustomWorkspace() throws Exception {
+        final String customWorkspaceValue = tmp.newFolder().getAbsolutePath();
         String customEnvVarName = "materialize_workspace_path";
         String customEnvVarValue = "${WORKSPACE}/materialize_workspace";
 
         FreeStyleProject project = j.createFreeStyleProject();
         project.setCustomWorkspace(customWorkspaceValue);
 
-        String propertiesContent = customEnvVarName + "=" + customEnvVarValue;
+        EnvVars.masterEnvVars.remove("WORKSPACE"); // ensure build node don't have such var already
 
-        EnvInjectJobPropertyInfo jobPropertyInfo = new EnvInjectJobPropertyInfo(null, propertiesContent, null, null, null, false);
         EnvInjectBuildWrapper envInjectBuildWrapper = new EnvInjectBuildWrapper();
-        envInjectBuildWrapper.setInfo(jobPropertyInfo);
+        envInjectBuildWrapper.setInfo(withPropContent(customEnvVarName + "=" + customEnvVarValue));
         project.getBuildWrappersList().add(envInjectBuildWrapper);
 
-        FreeStyleBuild build = project.scheduleBuild2(0).get();
-        Assert.assertEquals(Result.SUCCESS, build.getResult());
+        FreeStyleBuild build = j.buildAndAssertSuccess(project);
 
-        //1-- Compute expected injected var value
         //Retrieve build workspace
         String buildWorkspaceValue = build.getWorkspace().getRemote();
-        Assert.assertEquals(customWorkspaceValue, buildWorkspaceValue);
+        assertThat("1. Should see actual ws equal to custom", buildWorkspaceValue, equalTo(customWorkspaceValue));
+
         //Compute value with workspace
-        Map<String, String> mapEnvVars = new HashMap<String, String>();
-        mapEnvVars.put("WORKSPACE", buildWorkspaceValue);
-        String expectedCustomEnvVarValue = resolveVars(customEnvVarValue, mapEnvVars);
-
-        //2-- Get injected value for the specific variable
-        EnvInjectAction envInjectAction = build.getAction(EnvInjectAction.class);
-        Assert.assertNotNull(envInjectAction);
-        Map<String, String> envVars = envInjectAction.getEnvMap();
-        Assert.assertNotNull(envVars);
-        String resolvedValue = envVars.get(customEnvVarName);
-        Assert.assertNotNull(resolvedValue);
-
-        //3-- Test equals
-        Assert.assertEquals(expectedCustomEnvVarValue, resolvedValue);
+        String expectedCustomEnvVarValue = replaceMacro(customEnvVarValue,
+                of("WORKSPACE", buildWorkspaceValue));
+
+        assertThat("2. Property should be resolved with custom ws in build", build,
```

Comment:
Think this testcase is quite uncommon. And should be rewritten in future. So numbering is for quick understandig the place of error in this case.